Modules/_decimal/libmpdec/fourstep.c

```diff
@@ -186,13 +186,6 @@ four_step_fnt(mpd_uint_t *a, mpd_size_t n, int modnum)
         }
     }
 
-#if 0
```

Comment:
I think `libmpdec` should be left alone - it's a vendorised copy of an external library, so we should keep it in sync with the upstream version. Please could you revert the libmpdec changes?

Reply:
Similar comments may apply to the `libffi_osx` changes, though I'm less sure of the history there. Still, it's probably best to be conservative.

Reply:
> Please could you revert the libmpdec changes?

Done. 2.5.1 is published last year so the condition can't be easily written off as some artifact of an ancient version for ancient systems.

> Similar comments may apply to the libffi_osx change

I looked up for the current state of affairs in [libffi-3.4.2](https://github.com/libffi/libffi/tree/v3.4.2), both ifs are removed there:

- https://github.com/libffi/libffi/blob/v3.4.2/src/x86/ffi64.c#L161-L175 (`#if` would be on line 34)
- https://github.com/libffi/libffi/blob/v3.4.2/src/powerpc/ffi_darwin.c#L30-L35 (`#if` would be on line 176)

Indeed I can revert these two files and try to upgrade the whole vendored copy from 2.0 (circa 1998) to the latest version compatible with minimal CPython build environment. I don't know, though, how long it will take to trace through its changelog looking for relevant API changes.

Reply:
Thank you!

Reply:
The libmpdec change was reverted